packages/shared-ux/code_editor/mocks/monaco_mock/index.tsx

```diff
@@ -8,8 +8,10 @@
  */
 
 import React, { useEffect, KeyboardEventHandler } from 'react';
-import { type MonacoEditorProps } from 'react-monaco-editor';
 import { monaco } from '@kbn/monaco';
+// TODO: circular dependency
+// import type { MonacoEditorProps } from '@kbn/code-editor/react_monaco_editor';
+type MonacoEditorProps = any;
```

Comment:
this is a bummer, but I think it is an OK tradeoff as this seem to be private and doesn't affect consumers. Otherwise this would require a larger restructure and another package for `react-monaco-editor` folder and there is no need for that